src/contribute/runner-guide.md

```diff
@@ -0,0 +1,1103 @@
+---
+layout: default
+title: "Runner Guide"
+permalink: /contribute/runner-guide/
+---
+
+# Runner Guide
+
+This document is aimed at someone who has a data processing system and wants
+to use it to execute a Beam pipeline. It starts from the basics, to help you
+evaluate the work ahead. Then the sections become more and more detailed, to be
+a resource throughout the development of your runner.
+
+Topics covered:
+
+ * Basics of the Beam model, focused on what a runner author needs to know to
+   execute a pipeline
+ * Details of how to implement each of Beam's five primitives (of which two are
+   rather trivial and three are rather complex)
+ * Support code we provide for manipulating and executing pipelines
+ * How to test your runner
+ * How to make your runner play nice with SDKs
+ * The low-level Runner API protos used for cross-language pipelines
+
+* TOC
+{:toc}
+
+## Basics of the Beam model
+
+Suppose you have a data processing engine that can pretty easily process graphs
+of operations. You want to integrate it with the Beam ecosystem to get access
+to other languages, great event time processing, and a library of connectors.
+You need to know the core vocabulary:
+
+ * [_Pipeline_](#pipeline) - A pipeline is a graph of transformations that a user constructs
+   that defines the data processing they want to do.
+ * [_PCollection_](#pcollections) - Data being processed in a pipeline is part of a PCollection.
+ * [_PTransforms_](#ptransforms) - The operations executed within a pipeline. These are best
+   thought of as operations on PCollections.
+ * _SDK_ - A language-specific library for pipeline authors (we often call them
+   "users" even though we have many kinds of users) to build transforms,
+   construct their pipelines and submit them to a runner
+ * _Runner_ - You are going to write a piece of software called a runner that
+   takes a Beam pipeline and executes it using the capabilities of your data
+   processing engine.
+ 
+These concepts may be very similar to your processing engine's concepts. Since
+Beam's design is for cross-language operation and reusable libraries of
+transforms, there are some special features worth highlighting.
+
+### Pipeline
+
+A pipeline in Beam is a graph of PTransforms operating on PCollections. In the
+picture, the boxes are PTransforms and the arrows represent the contents of the
```

Comment:
I was going to paste in our very intro picture, but forgot. I don't think it adds much. Elsewhere pictures might be useful, but I don't imagine that a runner author needs to be reminded that it is a graph of transforms.